tests/queries/0_stateless/03512_bech32_enc_dec.sql

```diff
@@ -0,0 +1,150 @@
+-- baseline test, encode of value should match expected val
+select bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'));
+-- different hrp value should yield a different result
+select bech32Encode('tb', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'));
+-- exactly the max amount of characters (50) should work
+select bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6751e76e8199196d454941c45d1b3a323f1433bd6751e76e8199196d45494'));
+-- strange, but valid
+select bech32Encode('bcrt', unhex(''));
+
+-- test other hrps
+select bech32Encode('bcrt', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'));
+select bech32Encode('tltc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'));
+select bech32Encode('tltssdfsdvjnasdfnjkbhksdfasnbdfkljhaksdjfnakjsdhasdfnasdkfasdfasdfasdf', unhex('751e'));
+-- too many chars
+select bech32Encode('tltssdfsdvjnasdfnjkbhksdfasnbdfkljhaksdjfnakjsdhasdfnasdkfasdfasdfasdfdljsdfasdfahc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'));
+
+-- negative tests
+-- empty hrp
+select bech32Encode('', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'));
+select bech32Encode('bc', unhex('751E76E8199196D454941C45D1B3A323F1433BD6'));
+-- 51 chars should return nothing
+select bech32Encode('', unhex('751e76e8199196d454941c45d1b3a323f1433bd6751e76e8199196d454941c45d1b3a323f1433bd6751e76e8199196d45494a'));
+
+-- test with explicit witver = 0, should be same as default
+select bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'), 0) == 
+       bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'));
+
+-- testing bech32m algo
+select bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'), 1);
+select bech32Encode('tb', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'), 2);
+
+-- witversions >=1 should all be the same
+select bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'), 1) == 
+       bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'), 10);
+
+-- roundtrip
+select tup.1 as hrp, hex(tup.2) as data from (select bech32Decode(bech32Encode('bc', unhex('751e76e8199196d454941c45d1b3a323f1433bd6'))) as tup);
+
+CREATE TABLE hex_data
+(
+	hrp String,
+    data String,
+	witver UInt8
+)
+ENGINE = Memory;
+
+INSERT INTO hex_data VALUES
+('bc', '6687112a6eadb4d88d29c7a45da56eff0c23b0e14e757d408e', 0),
+('tb', '8f8cdd4364bb7dca11c49743da2c4b54062fa0388bbf924078', 1),
+('bc', '50b80d45cc275f36eb5fb2c22a93f6a4e83ba9380e55c67f6a', 15),
+('tb', 'b103a1937c6e2fb9de707a4be02d5d39e217b4bca7ce3c9c12', 0),
+('bcrt', '95eb334ff82ef8ad76151c29094abdae6c9e8bb8244523e347', 2);
```

Comment:
Please check if the following query works:
```
SELECT bech32Encode('bc', unhex(data)) FROM hex_data;
```

Reply:
Wow, I wouldn't have caught that, thanks! I have updated the code to handle mixed const and column data.